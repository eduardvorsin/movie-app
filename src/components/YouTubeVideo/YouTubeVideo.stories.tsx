import type { Meta, StoryObj } from '@storybook/react';
import YouTubeVideo, { Props } from './YouTubeVideo';
import { useTranslation } from 'react-i18next';

const meta: Meta<typeof YouTubeVideo> = {
	title: 'UI/YouTubeVideo',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the component'
		},
		testId: {
			description: 'Id for testing the component',
		},
		dictionary: {
			control: false,
			description: 'An object with keys for localization of the component',
		},
		width: {
			description: 'Width of the youtube video player',
		},
		height: {
			description: 'Height of the youtube video player',
		},
		videoId: {
			description: 'id for YouTube videos',
		},
		title: {
			description: 'text header for screen readers',
		},
		startTime: {
			description: 'The initial time when the video starts',
		},
		showControls: {
			description: 'Show controls in the YouTube player',
		},
		loading: {
			description: 'Allows you to add lazy loading for the poster',
		},
		posterQuality: {
			description: 'The quality of the YouTube poster',
		},
		muted: {
			description: 'Makes the default video without sound',
		},
		posterAlt: {
			description: 'Alternative text for the poster if it doesn\'t load',
		},
		posterSizes: {
			description: 'Dimensions of the YouTube video poster',
		},
	},
	component: YouTubeVideo,
	parameters: {
		docs: {
			description: {
				component: 'The YouTubeVideo component is a web interface element that provides user access to their account settings or personal preferences in the application.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof YouTubeVideo>;

const YouTubeVideoWithHooks = (props: Omit<Props, 'dictionary'>) => {
	const { t } = useTranslation('common');
	const dictionary = { playButton: t('youTubeVideo.button') };

	return (
		<YouTubeVideo
			{...props}
			dictionary={dictionary}
		/>
	);
}

export const Default: Story = {
	args: {
		className: 'max-w-[640px]',
		width: 640,
		height: 360,
		videoId: 'KLuTLF3x9sA',
		showControls: false,
	},
	parameters: {
		docs: {
			description: {
				story: 'The default state of the component',
			},
		},
	},
	render: (args) => (<YouTubeVideoWithHooks {...args} />),
};

export const StartTime: Story = {
	args: {
		className: 'max-w-[640px]',
		width: 640,
		height: 360,
		videoId: 'KLuTLF3x9sA',
		showControls: false,
		startTime: 50,
	},
	parameters: {
		docs: {
			description: {
				story: '',
			},
		},
	},
	render: (args) => (<YouTubeVideoWithHooks {...args} />),
};

export const ShowControls: Story = {
	args: {
		className: 'max-w-[640px]',
		width: 640,
		height: 360,
		videoId: 'KLuTLF3x9sA',
		showControls: true,
	},
	parameters: {
		docs: {
			description: {
				story: '',
			},
		},
	},
	render: (args) => (<YouTubeVideoWithHooks {...args} />),
};

export const Muted: Story = {
	args: {
		className: 'max-w-[640px]',
		width: 640,
		height: 360,
		videoId: 'KLuTLF3x9sA',
		muted: true,
	},
	parameters: {
		docs: {
			description: {
				story: '',
			},
		},
	},
	render: (args) => (<YouTubeVideoWithHooks {...args} />),
};