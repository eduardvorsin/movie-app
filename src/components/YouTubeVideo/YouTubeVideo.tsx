'use client';

import { useScreenWidth } from '@/hooks/useScreenWidth/useScreenWidth';
import { GeneralProps } from '@/types/shared';
import Image from 'next/image';
import { useEffect, useState } from 'react';
import ThemedImage from '../ThemedImage/ThemedImage';
import { Locales, fallbackLng, runsOnServerSide } from '@/i18n/settings';
import { useParams } from 'next/navigation';
import { useTranslation } from '@/i18n/client';

const youtubeVideoBasePath = 'https://www.youtube-nocookie.com/embed/';
const youtubePosterBasePath = 'https://i.ytimg.com/vi_webp/';

type Props = {
	width: number,
	height: number,
	videoId: string,
	title: string,
	startTime?: number,
	showControls?: boolean,
	loading?: 'eager' | 'lazy';
	posterQuality?: 'default' | 'mqdefault' | 'hqdefault' | 'sddefault' | 'maxresdefault',
	muted?: boolean,
	posterAlt: string,
	posterSizes?: string,
	testId?: string,
} & GeneralProps;

export default function YouTubeVideo({
	className,
	width,
	height,
	videoId,
	title,
	startTime,
	showControls = true,
	testId,
	loading,
	posterQuality,
	muted,
	posterAlt,
	posterSizes,
	...props
}: Props) {
	const lang = useParams()?.lang as Locales ?? fallbackLng;
	const { t } = useTranslation(lang);

	const [isIframeloading, setIsIframeLoading] = useState<boolean>(false);
	const screenWidth = useScreenWidth();
	const [posterResolution, setPosterResolution] = useState<Props['posterQuality']>(posterQuality ?? 'maxresdefault');

	const addIframe = () => setIsIframeLoading(true);

	const iframeSrc = new URL(videoId, youtubeVideoBasePath);
	iframeSrc.searchParams.append('rel', '0');
	iframeSrc.searchParams.append('autoplay', '1');
	iframeSrc.searchParams.append('muted', muted ? '1' : '0');

	if (startTime) {
		iframeSrc.searchParams.append('start', startTime.toString());
	}

	if (!showControls) {
		iframeSrc.searchParams.append('controls', '0');
	}

	const posterSrc = new URL(`${videoId}/${posterResolution}.webp`, youtubePosterBasePath);

	useEffect(() => {
		if (!runsOnServerSide) {
			setPosterResolution(posterQuality ?? screenWidth <= 768 ? 'hqdefault' : 'maxresdefault');
		}
	}, [posterQuality, screenWidth]);

	return (
		<div
			className={`relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 ${className}`}
			data-testid={testId}
			{...props}
		>
			{!isIframeloading && (
				<>
					<ThemedImage
						className='w-full object-cover'
						width={width}
						height={height}
						showSkeleton
						src={posterSrc.href}
						fallback={{
							light: '/assets/images/movie-card-placeholder-l-h.svg',
							dark: '/assets/images/movie-card-placeholder-d-h.svg'
						}}
						alt={posterAlt}
						loading={loading}
						sizes={posterSizes}
						style={{
							aspectRatio: `${width}/${height}`
						}}
					/>

					<button
						className='w-full h-full flex items-center justify-center absolute z-100 top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2'
						onClick={addIframe}
						aria-label={t('youTubeVideo.button')}
						title={t('youTubeVideo.button')}
					>
						<Image
							className='w-[4rem] h-[4rem]'
							src='/assets/icons/youtube_default.svg'
							alt=''
							width={64}
							height={64}
						/>
					</button>
				</>
			)}

			{isIframeloading && (
				<iframe
					className='w-full max-w-full h-auto'
					width={width}
					height={height}
					src={iframeSrc.href}
					title={title}
					allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share'
					allowFullScreen
					data-testid={testId}
					style={{
						aspectRatio: `${width}/${height}`
					}}
				/>
			)}
		</div>
	);
}


