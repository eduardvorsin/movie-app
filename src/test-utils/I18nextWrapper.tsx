import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;