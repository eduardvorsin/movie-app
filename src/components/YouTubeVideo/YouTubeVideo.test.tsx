import { render, screen } from "@testing-library/react";
import YouTubeVideo from "./YouTubeVideo";
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";
import userEvent from "@testing-library/user-event";

describe('YouTubeVideo tests', () => {
	it('is rendered correctly', async () => {
		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toBeInTheDocument();
	});

	it('is a basic snapshot', async () => {
		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});

	it('is a snapshot of when the video uploaded', async () => {
		const user = userEvent.setup();

		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});

	it('is a snapshot when showControls is false', async () => {
		const user = userEvent.setup();

		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				showControls={false}
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});

	it('is a snapshot when muted is true', async () => {
		const user = userEvent.setup();

		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				muted
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});

	it('is a snapshot when the value for startTime is set', async () => {
		const user = userEvent.setup();

		const dictionary = { playButton: 'To watch' };
		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				startTime={50}
				dictionary={dictionary}
				testId='test-youtube-video'
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});
});

describe('YouTubeVideo integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('youTubeVideo.playButton'));
		const dictionary = { playButton: i18next.t('youTubeVideo.playButton') };

		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(await screen.findByRole<HTMLButtonElement>('button', { name: translation })).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('youTubeVideo.playButton'));
		const dictionary = { playButton: i18next.t('youTubeVideo.playButton') };

		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(await screen.findByRole<HTMLButtonElement>('button', { name: translation })).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');
		const dictionary = { playButton: i18next.t('youTubeVideo.playButton') };

		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');
		const dictionary = { playButton: i18next.t('youTubeVideo.playButton') };

		render(
			<YouTubeVideo
				width={386}
				height={217}
				key={'KLuTLF3x9sA'}
				videoId={'KLuTLF3x9sA'}
				title='test title'
				loading='lazy'
				posterAlt='test-youtube-video'
				dictionary={dictionary}
				testId='test-youtube-video'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-youtube-video')).toMatchSnapshot();
	});
});