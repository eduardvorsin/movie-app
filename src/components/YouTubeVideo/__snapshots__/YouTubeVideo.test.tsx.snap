// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTubeVideo integration tests is a snapshot with English localization 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <div
    class="relative w-full object-cover"
  >
    <div
      aria-hidden="true"
      class="bg-neutral-400 dark:bg-dark-neutral-300 max-w-full transition-colors duration-150  [&]:absolute top-0 left-0 z-100 relative overflow-hidden isolate before:absolute before:inset-0 before:-translate-x-full before:animate-skeleton before:bg-gradient-to-r before:from-transparent before:via-neutral-0/50 dark:before:via-neutral-0/25 before:to-transparent before:transition-colors before:duration-150"
      style="width: 100%; height: 100%;"
    />
    <img
      alt="test-youtube-video"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="http://localhost/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <button
    aria-label="To watch"
    class="w-full h-full flex items-center justify-center absolute z-100 top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
    title="To watch"
  >
    <img
      alt=""
      class="w-[4rem] h-[4rem]"
      data-nimg="1"
      decoding="async"
      height="64"
      loading="lazy"
      src="/assets/icons/youtube_default.svg"
      style="color: transparent;"
      width="64"
    />
  </button>
</div>
`;

exports[`YouTubeVideo integration tests is a snapshot with Russian localization 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <div
    class="relative w-full object-cover"
  >
    <div
      aria-hidden="true"
      class="bg-neutral-400 dark:bg-dark-neutral-300 max-w-full transition-colors duration-150  [&]:absolute top-0 left-0 z-100 relative overflow-hidden isolate before:absolute before:inset-0 before:-translate-x-full before:animate-skeleton before:bg-gradient-to-r before:from-transparent before:via-neutral-0/50 dark:before:via-neutral-0/25 before:to-transparent before:transition-colors before:duration-150"
      style="width: 100%; height: 100%;"
    />
    <img
      alt="test-youtube-video"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="http://localhost/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <button
    aria-label="Смотреть"
    class="w-full h-full flex items-center justify-center absolute z-100 top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
    title="Смотреть"
  >
    <img
      alt=""
      class="w-[4rem] h-[4rem]"
      data-nimg="1"
      decoding="async"
      height="64"
      loading="lazy"
      src="/assets/icons/youtube_default.svg"
      style="color: transparent;"
      width="64"
    />
  </button>
</div>
`;

exports[`YouTubeVideo tests is a basic snapshot 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <div
    class="relative w-full object-cover"
  >
    <div
      aria-hidden="true"
      class="bg-neutral-400 dark:bg-dark-neutral-300 max-w-full transition-colors duration-150  [&]:absolute top-0 left-0 z-100 relative overflow-hidden isolate before:absolute before:inset-0 before:-translate-x-full before:animate-skeleton before:bg-gradient-to-r before:from-transparent before:via-neutral-0/50 dark:before:via-neutral-0/25 before:to-transparent before:transition-colors before:duration-150"
      style="width: 100%; height: 100%;"
    />
    <img
      alt="test-youtube-video"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="http://localhost/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ytimg.com%2Fvi_webp%2FKLuTLF3x9sA%2Fhqdefault.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <button
    aria-label="To watch"
    class="w-full h-full flex items-center justify-center absolute z-100 top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
    title="To watch"
  >
    <img
      alt=""
      class="w-[4rem] h-[4rem]"
      data-nimg="1"
      decoding="async"
      height="64"
      loading="lazy"
      src="/assets/icons/youtube_default.svg"
      style="color: transparent;"
      width="64"
    />
  </button>
</div>
`;

exports[`YouTubeVideo tests is a snapshot of when the video uploaded 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class="w-full max-w-full h-auto"
    height="217"
    src="https://www.youtube-nocookie.com/embed/KLuTLF3x9sA?rel=0&autoplay=1&mute=0"
    title="test title"
    width="386"
  />
</div>
`;

exports[`YouTubeVideo tests is a snapshot when muted is true 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class="w-full max-w-full h-auto"
    height="217"
    src="https://www.youtube-nocookie.com/embed/KLuTLF3x9sA?rel=0&autoplay=1&mute=1"
    title="test title"
    width="386"
  />
</div>
`;

exports[`YouTubeVideo tests is a snapshot when showControls is false 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class="w-full max-w-full h-auto"
    height="217"
    src="https://www.youtube-nocookie.com/embed/KLuTLF3x9sA?rel=0&autoplay=1&mute=0&controls=0"
    title="test title"
    width="386"
  />
</div>
`;

exports[`YouTubeVideo tests is a snapshot when the value for startTime is set 1`] = `
<div
  class="relative dark:bg-dark-neutral-250 bg-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-youtube-video"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class="w-full max-w-full h-auto"
    height="217"
    src="https://www.youtube-nocookie.com/embed/KLuTLF3x9sA?rel=0&autoplay=1&mute=0&start=50"
    title="test title"
    width="386"
  />
</div>
`;
